force-app/main/default/classes/CheckingGenderWhenContactCreatedUtility.cls

```diff
@@ -0,0 +1,34 @@
+public with sharing class CheckingGenderWhenContactCreatedUtility {
+    public static void  detectGender(List<Contact> contacts){
+
+
+        For(Contact contact:contacts){
+
+            if(checkLastChar(contact.FirstName)==true){
+                contact.Gender__c='female';
+                }
+            else
+                {
+                contact.Gender__c='male';
+                }
+            System.Debug(contact.Gender__c);
+        }
+    }
+    
+    
+        public static Boolean checkLastChar(String firstName){
+            String toCheck = firstName;
+        
+            if(toCheck.substring(toCheck.length() - 1)=='a')
+                {
```

Comment:
I tak samo klamra otwierająca, to sie tyczy wszystkich ifów i klamr, to co tam wyżej pisałem :D